client/packages/common/src/authentication/hooks/usePermissionCheck.ts

```diff
@@ -23,6 +26,10 @@ export const usePermissionCheck = (permission: UserPermission) => {
   }, [error]);
 
   if (previous.current === AuthError.PermissionDenied && !error) {
-    navigate(RouteBuilder.create(AppRoute.Dashboard).build());
+    if (onPermissionDenied) {
+      onPermissionDenied();
```

Comment:
Using this to close the modal, but now that I have the other component could just conditionally render the modal depending on the permission denied check?

Reply:
Hm yeah, I think it would feel nicer to see _either_ the modal or the permission denied message, rather than one on top of the other